### ISSUE TITLE
Fallback at httpRequest.responseXML instanceof Document check

### ISSUE THREAD

**Issue by welcomemax** [tanem/svg-injector#1192]
Hello tahem!

We've run into one very nasty conflict on several client sites where we use the SVG Injector.

They use a very very ancient library called "mootools" with this interesting method that overriding window.Document (such a shame). https://mootools.net/core/docs/1.6.0/Element/Element#Window:document-id

![image](https://user-images.githubusercontent.com/5702867/209154405-0ababb9f-34e5-4ca0-b661-059e4c36330c.png)

As a consequence, the `instanceof Document` check fails. ([load-svg-cached.ts](https://github.com/tanem/svg-injector/blob/master/src/load-svg-cached.ts) and [load-svg-uncached.ts](https://github.com/tanem/svg-injector/blob/master/src/load-svg-uncached.ts))

From the user-land side, we cannot influence this in any way and are thinking about a stub.

Could you add some fallback if this check fails on your end?

**Comment by welcomemax** [comment]
In any case, the else if check looks insufficient without a default return or else.

How about such a handler?

```ts
const isValidSVGDocument = (document: unknown) => {
  return
    httpRequest.responseXML instanceof Document &&
    httpRequest.responseXML.documentElement &&
    httpRequest.responseXML.documentElement instanceof SVGSVGElement
}

const loadSvgCached = (
  url: string,
  httpRequestWithCredentials: boolean,
  callback: Errback
) => {
  ...
  makeAjaxRequest(url, httpRequestWithCredentials, (error, httpRequest) => {
    if (error) {
      cache.set(url, error);
    } 
    
    if (isValidSVGDocument(httpRequest.responseXML)) {
      cache.set(url, httpRequest.responseXML.documentElement);
    } else {
      cache.set(url, 'NOT_VALID_XML_RESPONSE');
    }

    processRequestQueue(url);
  })
}
```

**Comment by tanem** [comment]
Hey @welcomemax, thanks for raising this issue. Yea working with these older libs can be tricky huh 😅

Since `httpRequest.responseXML` should be [either `Document` or `null`](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseXML), I think we can get away with simplifying the check so your use-case will actually work. [Here's a proposal](https://github.com/tanem/svg-injector/pull/1207), keen to hear your thoughts?

Have also hacked up a CodeSandbox example of the fix [here](https://codesandbox.io/s/nameless-pine-gc3o29). If we end up merging the PR I'll make it a proper Live Example.

**Comment by tanem** [comment]
Example added. I'll merge and release this PR now. Feel free to comment on this issue though if you have any further feedback 🙏

**Comment by tanem** [comment]
Released in `v10.1.44`.

**Comment by welcomemax** [comment]
Thank you a lot, @tanem! Really appreciate!
